### PR TITLE
PLANET-4023 Allow "Take Action" boxout in Posts 

### PR DESCRIPTION
**Pull request by dpivo**
- Strip the take action boxout block from the post content and add it to where the current shortcode would be added.

### CODE REVIEW

File: single.php [hunk 1]
```diff
@@ -21,6 +21,16 @@
 $post            = Timber::query_post( false, 'P4_Post' );
 $context['post'] = $post;
 
+// Strip Take Action Boxout block from the post content to add outside the normal block container.
+if ( strpos( $post->post_content, '[shortcake_take_action_boxout' ) ) {
```
Comment (kirdia) [inline single.php]: better check if strpos !== false, cause strpos could return 0 and this won't be evaluated

Comment (dpivo) [reply single.php]: Good point! Changed...

File: single.php [hunk 2]
```diff
@@ -21,6 +21,16 @@
 $post            = Timber::query_post( false, 'P4_Post' );
 $context['post'] = $post;
 
+// Strip Take Action Boxout block from the post content to add outside the normal block container.
+if ( false !== strpos( $post->post_content, '[shortcake_take_action_boxout' ) ) {
+
+	$take_action_boxout_block_start = strpos( $post->post_content, '[shortcake_take_action_boxout' );
+	$take_action_boxout_block_end   = strpos( $post->post_content, ']', $take_action_boxout_block_start );
+	$take_action_boxout_block       = substr( $post->post_content, $take_action_boxout_block_start, $take_action_boxout_block_end );
```
Comment (kirdia) [inline single.php]: this won't work. substr needs length of the string as the 3rd parameter not an index.

Comment (dpivo) [reply single.php]: calculated the length and added used that instead.

**Pull request merged**